### PR TITLE
Only ignored assets

### PR DESCRIPTION
**Pull request by nebolax**

### CODE REVIEW

File: rotkehlchen/api/v1/schemas.py [hunk 1]
```diff
@@ -1757,6 +1757,7 @@ class AssetsPostSchema(DBPaginationSchema, DBOrderBySchema):
     symbol = fields.String(load_default=None)
     asset_type = SerializableEnumField(enum_class=AssetType, load_default=None)
     include_ignored_assets = fields.Boolean(load_default=True)
+    only_ingored_assets = fields.Boolean(load_default=False)
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1760]:
```suggestion
    only_ignored_assets = fields.Boolean(load_default=False)
```

File: rotkehlchen/api/v1/schemas.py [hunk 2]
```diff
@@ -1776,6 +1777,11 @@ def validate_schema(  # pylint: disable=no-self-use
                 message='Multiple fields ordering is not allowed.',
                 field_name='order_by_attributes',
             )
+        if data['include_ignored_assets'] is False and data['only_ingored_assets'] is True:
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1780]:
```suggestion
        if data['include_ignored_assets'] is False and data['only_ignored_assets'] is True:
```

File: rotkehlchen/api/v1/schemas.py [hunk 3]
```diff
@@ -1776,6 +1777,11 @@ def validate_schema(  # pylint: disable=no-self-use
...
+            raise ValidationError(
+                message='Cannot use only_ingored_assets=True with include_ignored_assets=False',
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1782]:
```suggestion
                message='Cannot use only_ignored_assets=True with include_ignored_assets=False',
```

File: rotkehlchen/api/v1/schemas.py [hunk 4]
```diff
@@ -1785,7 +1791,13 @@ def make_assets_post_query(
     ) -> Dict[str, Any]:
         ignored_assets_identifiers = None
         with self.db.user_write() as write_cursor, GlobalDBHandler().conn.read_ctx() as globaldb_read_cursor:  # noqa: E501
-            if data['include_ignored_assets'] is False:
+            ignored_assets_filter: Optional[Literal['only_ignored', 'exclude_ignored']] = None
+            if data['only_ingored_assets'] is True:
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1795]:
```suggestion
            if data['only_ignored_assets'] is True:
```

File: rotkehlchen/db/filtering.py [hunk 5]
```diff
@@ -1000,12 +1001,19 @@ def make(
                 column='identifier',
                 values=identifiers,
             ))
-        if ignored_assets_identifiers is not None:
+        if ignored_assets_filter == 'only_ignored':
```
Comment (LefterisJP) [inline rotkehlchen/db/filtering.py:1004]: you can probably make the code shorter and more readable here and avoid extra if checks and type ignores

`ignored_assets_operator` is the operator you will pass directly in the function. Either `IN` or `NOT IN`

```python
if ignored_assets_identifiers is not None:
            filters.append(DBMultiStringFilter(
                and_op=True,
                column='identifier',
                operator=ignored_assets_operator,
                values=ignored_assets_identifiers,
            ))

File: rotkehlchen/db/filtering.py [hunk 6]
```diff
@@ -949,6 +949,7 @@ def make(
             identifiers: Optional[List[str]] = None,
             return_exact_matches: bool = False,
             evm_chain: Optional[ChainID] = None,
+            ignored_assets_filter: Optional[Literal['only_ignored', 'exclude_ignored']] = None,
```
Comment (LefterisJP) [inline rotkehlchen/db/filtering.py:952]:
```suggestion
            ignored_assets_operator: Optional[Literal['IN', 'NOT IN']] = None,
```

File: rotkehlchen/api/v1/schemas.py [hunk 7]
```diff
@@ -1785,7 +1791,13 @@ def make_assets_post_query(
...
+                ignored_assets_filter = 'only_ignored'
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1796]: here use the operator directly as the value. Also can make it shorter with the ternary operator

**Review by LefterisJP — COMMENTED** [review]
you will also need to add this new flag to the api docs

File: rotkehlchen/api/v1/schemas.py (hunk 7 above)
Comment (nebolax) [reply rotkehlchen/api/v1/schemas.py:1796]: I made it like this for better readability

Comment (LefterisJP) [reply rotkehlchen/api/v1/schemas.py:1796]: It's not. Please do what I asked

**Comment by lukicenturi** [comment]
Functionality works

File: rotkehlchen/db/filtering.py [hunk 8]
```diff
@@ -1004,7 +1005,7 @@ def make(
             filters.append(DBMultiStringFilter(
                 and_op=True,
                 column='identifier',
-                operator='NOT IN',
+                operator=ignored_assets_operator,  # type: ignore  # ignored_assets_operator is definitely not None  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/db/filtering.py:1008]: I think you may be able to get rid of the type ignore here if you use @overload to specify the relation that when calling with `ignored_assets_identifiers` being not None you always have an operator

**Review by LefterisJP — COMMENTED** [review]
Left some more comments.
You still have not done all I asked you in the last review. The api docs are still not adjusted.

**Review by LefterisJP — APPROVED** [review]
LGTM. Frontend tests will be adjusted by Luki in the frontend PR

**Pull request merged**